### PR TITLE
Add pullquote tranformations

### LINKED ISSUE #14159: Allow Paragraph-Pullquote Conversion/Transformation
**Is your feature request related to a problem? Please describe.**
I expected that I could convert a Paragraph to a Pullquote. I especially expect this because I can convert a pullquote to a blockquote and a blockquote to a paragraph.

**Describe the solution you'd like**
Standardize the block-type conversions. If I can convert a paragraph to a blockquote and vice-versa, I'd expect both block types to be convertible to the same list of blocks. However, that is not the case.

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description
Resolves #14159

Now,  the `Pullquote` transformations are handled in an inconsistent way compared to a `Quote` block, that is quite similar.

For example you can achieve transformation from a `Paragraph` to `Pullquote`, by first transforming the `Paragraph` to a `Quote` and then the `Quote` to a `Pullquote`.

This PR will allow `Pullquote` block to be transformed from and to all the available blocks that are allowed for `Quote` block.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/pullquote/transforms.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+			isMultiBlock: true,
+			blocks: [ 'core/paragraph' ],
+			transform: ( attributes ) => {
+				return createBlock( 'core/pullquote', {
+					value: toHTMLString( {
+						value: join(
+							attributes.map( ( { content } ) =>
+								create( { html: content } )
+							),
+							'\u2028'
```
Comment (mcsf) [inline packages/block-library/src/pullquote/transforms.js:20]: I don't feel confident that this is the optimal way to do it, so I'll defer to @ellatrix for review. This includes the logic further down in the `'to'` transforms.

File: packages/block-library/src/list/transforms.js [hunk 2]
```diff
@@ -78,6 +78,19 @@ const transforms = {
 				} );
 			},
 		},
+		{
+			type: 'block',
+			blocks: [ 'core/pullquote' ],
```
[the remark this answers is not in the document]
Comment (ntsekouras) [reply packages/block-library/src/list/transforms.js:83]: You are so right Miguel! Changed it

File: packages/block-library/src/pullquote/transforms.js (hunk 1 above)
Comment (ntsekouras) [reply packages/block-library/src/pullquote/transforms.js:20]: This was actually the same as `Quote`. Didn't really dig into that. Seemed to me a Business Logic decision that has been made, to transform the first part only..

Comment (mcsf) [reply packages/block-library/src/pullquote/transforms.js:20]: You're right. That's good enough for me!

**Review by mcsf — APPROVED** [review]
This does highlight the potential for Pullquote to be a block variation of Quote, but that's something to consider some other time.

LGTM!

**Comment by ntsekouras** [comment]
Thanks @mcsf ! I totally agree that these two are so similar that could be handled together in the future.

**Pull request merged**